nodejs/week4/spam-detector/index.js

```diff
@@ -0,0 +1,89 @@
+class Email {
+	constructor(subject, body) {
+		this.subject = subject;
+		this.body = body;
+	}
+}
+
+class SpamDetector {
+	badWordCheck(email) {
+		const badWords = ["viagra", "offer", "free", "business proposal"];
+
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split(" ");
+
+		const badWordsInEmail = emailBodyIntoArray.filter((item) => {
+			return badWords.includes(item.toLowerCase());
+		});
+
+		if (badWordsInEmail.length !== 0) {
+			return true;
+		} else {
+			return false;
+		}
+	}
+
+	subjectCheck(email) {
+		const emailSubject = email.subject;
+
+		if (emailSubject.toLowerCase() === "hello") {
+			return true;
+		} else {
+			return false;
+		}
+	}
+	upperCaseCheck(email) {
+		const emailBody = email.body;
+		const emailBodyIntoArray = emailBody.split("");
+		const upperCaseArray = emailBodyIntoArray.filter((words) => {
+			return words == words.toUpperCase();
+		});
+
+		const percentageCheck =
+			(upperCaseArray.length * 100) / emailBodyIntoArray.length;
+
+		if (percentageCheck > 60) {
+			return true;
+		} else {
+			return false;
+		}
```

Comment:
really understand why i can do this without if else

```suggestion
		return percentageCheck > 60
```

Reply:
I think it is because I gave percentage check as a variable and that variable in the hasMostlyUpperCaseCharacters function scope so I can directly return it with a condition.